multiaddr_test.go

```diff
@@ -558,6 +575,31 @@ func TestInvalidP2PAddr(t *testing.T) {
 	}
 }
 
+func TestInvalidP2PAddrString(t *testing.T) {
+	hashedData, err := mh.Sum([]byte("test"), mh.SHA2_256, -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// using MD5 since it's not a valid data codec
+	unknownCodecCID := cid.NewCidV1(mh.MD5, hashedData).String()
+
+	badStringAddrs := []string{
+		"/p2p/k2k4r8oqamigqdo6o7hsbfwd45y70oyynp98usk7zmyfrzpqxh1pohl-", // invalid multibase encoding
+		"/p2p/?unknownmultibase", // invalid multibase encoding
```

Comment:
Is this a safe/reasonable test? I guess we can just remove it if we ever allocate the `?` multibase and this test fails.

Reply:
Seems reasonable.